File: TheComfortZone.DTO/User/UserUpdateRequest.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Text;

namespace TheComfortZone.DTO.User
{
    public class UserUpdateRequest
    {
        [Required(AllowEmptyStrings = false)]
        [MaxLength(30)]
        public string FirstName { get; set; }
        [Required(AllowEmptyStrings = false)]
        [MaxLength(30)]
        public string LastName { get; set; }
        [Required(AllowEmptyStrings = false)]
        [MaxLength(70)]
        public string Adress { get; set; }
        [Required(AllowEmptyStrings = false)]
        [MaxLength(15)]
        [Phone]
        public string PhoneNumber { get; set; }
        [Required(AllowEmptyStrings = false)]
        [MaxLength(30)]
        [EmailAddress]
        public string Email { get; set; }
        [Required(AllowEmptyStrings = false)]
        [MaxLength(20)]
        public string Username { get; set; }
        [MaxLength(20)]
        public string Password { get; set; }
        [MaxLength(20)]
        public string PasswordConfirmation { get; set; }
    }
}

File: the_comfort_zone_mobile/lib/model/user/user_update_request.dart
import 'package:json_annotation/json_annotation.dart';
part 'user_update_request.g.dart';

@JsonSerializable()
class UserUpdateRequest {
  String? firstName;
  String? lastName;
  String? username;
  String? adress;
  String? phoneNumber;
  String? email;
  String? password;
  String? passwordConfirmation;

  UserUpdateRequest();

  factory UserUpdateRequest.fromJson(Map<String, dynamic> json) => _$UserUpdateRequestFromJson(json);
  /// Connect the generated [_$UserUpdateRequestToJson] function to the `toJson` method.
  Map<String, dynamic> toJson() => _$UserUpdateRequestToJson(this);
}
